virttest/libvirt_xml/base.py

```diff
@@ -436,3 +437,93 @@ def load_xml_module(path, name, type_list):
             "Can't find class %s in %s module in "
             "%s: %s" % (name.capitalize(), name, path, str(detail))
         )
+
+
+class VMIDMapXML(LibvirtXMLBase):
```

Comment:
I'd like to suggest that we move this class to a new file, which could similar to [character.py ](https://github.com/avocado-framework/avocado-vt/blob/c1c726a17b49c9890a44ee32ca00b96563483baa/virttest/libvirt_xml/devices/character.py)

Reply:
@chloerh AFAIK, "Character" is an actual base class for a device such as serial. But here the 'idmap' is not a device but a device subelement that is used by virtiofs and domain xml. Are you sure we should move this to a specific file idmap.py? I just want to double check because it seems incoherent for me.

Reply:
Actually the address element, common to all devices, is in devices.base. Given that idmap seems to be available on domain level, I think libvirt_xml.base is a good place for it.

Reply:
Oh, I understand your concern of putting it to devices/ because 'idmap' is not a device. But I also have this intuition that libvirt_xml.base used to act as a module to handle universal behaviors of all libvirt_xml subclasses, which seems not quite fit for one xml class. Maybe we could put it next to libvirt_xml/network_xml?

Reply:
I can do that if you insist but I have my doubts: all xml classes on the libvirt_xml level refer to a libvirt XML class, network xml is not a device but it's its own XML, not related to domain or subelements. The same holds for the other files: backup_xml, checkpoint_xml, snapshot_xml, vol_xml, aso.
The only exception I've found: sysinfo which is a subelement of vm_xml and I would actually expect it to live in that file.
I think your intuition about base.py scope I think it's right. So maybe what's missing is a subfolder 'misc' where both sysinfo_xml and idmap_xml should live?  Or we could create a file domaincommon_xml where we put both? (I got the idea from domaincommon.rng, where idmap is defined both for the domain as well as for the filesystem.)

Reply:
I like the idea of creating a subfolder for sysinfo_xml and idmap_xml, which could clarify the structure of all libvirt-XML classes. And I'm also ok with the domaincommon_xml idea, just a little worry that the file might become large in the future (which seem unlikely).  It's your call. Thank you for coming up with these great solutions!

Reply:
Okay, I created the 'other' subfolder. Btw I found that sysinfo was indeed already defined so I removed the sysinfo_xml file and point the reference to that instance. I'll run tests covering the idmap and sysinfo first before I re-request your review though.

Reply:
@chloerh I updated the code as required. I rerun some tests for this last change, including the sysinfo change: https://github.com/autotest/tp-libvirt/pull/5590#issuecomment-2109694624

The original class `SysinfoXML` was only used in tp-libvirt's virsh_sysinfo.